src/util/util_creds.h

```diff
@@ -64,12 +64,25 @@ typedef void * SEC_CTX;
 #endif /* done HAVE_SELINUX */
 
 #include <sys/socket.h>
+#ifdef __FreeBSD__
+#include <sys/param.h>
+#include <sys/ucred.h>
```

Comment:
I think
https://github.com/SSSD/sssd/blob/e76849bab7502ea8a3971b3b82c89389e91d3dee/src/external/platform.m4#L46
should be extended to detect availability of `xucred`